notebooks/12-xarray-advanced.md

```diff
@@ -0,0 +1,686 @@
+    
+$T_{^{\circ}C} = T_{K} - 273.15$
+    
+Create a histogram of the `temperature_c` to check the distribution of all the temperature valus in the data set. Use an appropriate number of bins to draw the histogram.
+
+<details><summary>Hints</summary>
+
+* Xarray - similar to Numpy - applies the mathematical operation element-wise, so no need for loops.
+* Most plot functions work on `DataArray`, so make sure to select the variable to apply the `.plot.hist()`.
+* One can define the number of bins using the `bins` parameter in the `hist` method.
+
+</details>    
+    
+</div>
+
+```{code-cell} ipython3
+:tags: [nbtutor-solution]
+
+era5_renamed["temperature_c"] = era5_renamed["temperature_k"] - 273.15
+```
+
+```{code-cell} ipython3
+:tags: [nbtutor-solution]
+
+era5_renamed["temperature_c"].plot.hist(bins=50);
+```
+
+<div class="alert alert-success">
+
+**EXERCISE**:
+
+Calculate the total snowfall of the entire region of the dataset in function of time and create a line plot showing total snowfall in the y-axis and time in the x-axis.
+
+<details><summary>Hints</summary>
+
+* You need to calculate the total (`sum`) snowfall (`snowfall_m`) in function of time, i.e. aggregate over both the `longitude` and `latitude` dimensions (`dim=["latitude", "longitude"]`).
+* As the result is a DataArray with a single dimension, the default `plot` will show a line, but you can be more explicit by saying `.plot.line()`.
+
+</details>    
+    
+</div>
+
+```{code-cell} ipython3
+:tags: [nbtutor-solution]
+
+era5_renamed["snowfall_m"].sum(dim=["latitude", "longitude"]).plot.line(figsize=(18, 5))
+```
+
+<div class="alert alert-success">
+
+**EXERCISE**:
+
+The speed to sound is temperature linearly dependent:
+    
+$v = 331.5 + (0.6 \cdot T)$
+    
+with $v$ the speed of light and $T$ the temperature in degrees celsius.    
+    
+Add a new variable to the `era5_renamed` data set, called `speed_of_light_m_s`, that calculates for each location and each time stamp in the data set the temperature corrected speed of light.
+    
+Create a scatter plot to check the (linear) relationship you just calculated by comparing all the `speed_of_light_m_s` and `temperature_c` data points in the data set.
+
+<details><summary>Hints</summary>
+
+* Creating a new variable is similar to GeoPandas/Pandas/dictionaries, `ds["MY_NEW_VAR"] = ...`.
+* Remember, calculations are __element-wise__ just as in Numpy/Pandas; so no need for loops. The numbers (331.5, 0.6) are broadcasted to all elements in the data set to do the calculation.
+* To compare two variables in a data set visually, `plot.scatter()` it is.
+
+</details>    
+    
+</div>
+
+```{code-cell} ipython3
+:tags: [nbtutor-solution]
+
+era5_renamed["speed_of_light_m_s"] = 331.5 + (0.6*era5_renamed["temperature_c"])
+```
+
+```{code-cell} ipython3
+:tags: [nbtutor-solution]
+
+era5_renamed.plot.scatter("temperature_c", "speed_of_light_m_s", s=1, alpha=0.1)
+```
+
```

Comment:
Something like `era5_renamed.sum(dim="time").plot.scatter("temperature_c", "precipitation_m")` actually does give a clear correlation and thus might also be useful to add as exercise with a scatterplot (and it gives a bit more realistic scatter plot, as the one above is of course a linear line, since you just calculated it like that)

Reply:
I propose to leave it as such. I just changed the wording to `Create a scatter plot to control the (linear) relationship you just calculated`. I do think an additional exercise is not required.